app/src/main/index.ts

```diff
@@ -30,6 +30,10 @@ import { syncMetadata } from "./sync";
 import workerPath from "./fetch/worker?modulePath";
 import { Lock } from "./sync/lock";
 import { Config } from "./config";
+import { setUmask } from "./umask";
+
+// Set umask so any files written are owner-only read/write (600).
+setUmask();
```

Comment:
nice! maybe we can add a comment note here that we need to set this before spawning the worker thread to avoid potential race: https://github.com/nodejs/node/issues/32321 

in our usage I don't foresee this being an issue, but maybe worth noting explicitly

Reply:
Can do.